subsys/usb/device/class/netusb/function_rndis.c

```diff
@@ -62,9 +62,9 @@ USBD_CLASS_DESCR_DEFINE(primary, 0) struct usb_rndis_config rndis_cfg = {
 		.bDescriptorType = USB_DESC_INTERFACE_ASSOC,
 		.bFirstInterface = 0,
 		.bInterfaceCount = 0x02,
-		.bFunctionClass = USB_BCC_CDC_CONTROL,
-		.bFunctionSubClass = 6,
-		.bFunctionProtocol = 0,
+		.bFunctionClass = USB_BCC_MISCELLANEOUS,
+		.bFunctionSubClass = 4,
+		.bFunctionProtocol = 1,
```

Comment:
The issue with that is, if it is not a CDC class then why should it use CDC descriptors? 
I guess all CDC descriptors below can be omitted, like https://learn.microsoft.com/en-us/azure/rtos/usbx/usbx-device-stack-supplemental-2, still weird.

Reply:
I am not sure why Microsoft specifies completely different descriptors in that example (_it was noted that in the discussion as well_).

However, such descriptors won't allow Windows to load correct drivers as noted by the screenshots above. At least on Windows 10. Not sure about Windows 11.

Reply:
Tested with descriptors provided in the example:

![ms_descriptors](https://user-images.githubusercontent.com/5956504/218046481-d097910e-b56c-4848-a958-39e924fd4bfb.JPG)

Windows won't load RNDIS driver automatically. They also mention:

> On windows platforms a .inf file needs to be present on the host that matches the device descriptor.

But using descriptors as specified by USB-IF or by Microsoft in "Supported Device Classes" (_Class: Miscellaneous (EFh), SubClass: 04h, Protocol: 01h_) everything works automatically. Windows loads RNDIS driver.

Reply:
> The issue with that is, if it is not a CDC class then why should it use CDC descriptors?

Why do you think this is CDC descriptor? Class 0xEF, SubClass 0x04, Protocol 0x01 is RNDIS over Ethernet. See https://www.usb.org/defined-class-codes

Reply:
@tmon-nordic he probably refers to the Microsoft example https://learn.microsoft.com/en-us/azure/rtos/usbx/usbx-device-stack-supplemental-2

Reply:
> > The issue with that is, if it is not a CDC class then why should it use CDC descriptors?
> 
> Why do you think this is CDC descriptor? Class 0xEF, SubClass 0x04, Protocol 0x01 is RNDIS over Ethernet. See https://www.usb.org/defined-class-codes

Our is CDC descriptor. Anyway, I think we should just align it with example in https://learn.microsoft.com/en-us/azure/rtos/usbx/usbx-device-stack-supplemental-2 and remove CDC descriptor. @GeorgeCGV can you please test if it also works for you?

```diff
diff --git a/subsys/usb/device/class/netusb/function_rndis.c b/subsys/usb/device/class/netusb/function_rndis.c
index cef31a6a718..8d263563f0c 100644
--- a/subsys/usb/device/class/netusb/function_rndis.c
+++ b/subsys/usb/device/class/netusb/function_rndis.c
@@ -45,10 +45,6 @@ static struct k_thread cmd_thread_data;
 struct usb_rndis_config {
        struct usb_association_descriptor iad;
        struct usb_if_descriptor if0;
-       struct cdc_header_descriptor if0_header;
-       struct cdc_cm_descriptor if0_cm;
-       struct cdc_acm_descriptor if0_acm;
-       struct cdc_union_descriptor if0_union;
        struct usb_ep_descriptor if0_int_ep;
 
```
        struct usb_if_descriptor if1;

Reply:
@jfischer-no still works on windows with removed `cdc_header_descriptor`, `cdc_cm_descriptor`, `cdc_acm_descriptor`, and `cdc_union_descriptor` while using USB-IF defined class for RNDIS Ethernet.

Reply:
> Should I include that in this PR?

Yes, also please add a note to commit message that RNDIS implementation is not a CDC class.